File: stepik-union/src/main/java/org/stepik/plugin/projectView/PyCharmNavBarModelExtension.java
package org.stepik.plugin.projectView;

import com.intellij.ide.navigationToolbar.DefaultNavBarExtension;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.Nullable;

/**
 * @author meanmail
 */
public class PyCharmNavBarModelExtension extends DefaultNavBarExtension {
    @Nullable
    @Override
    public String getPresentableText(@Nullable final Object object) {
        String text = NavBarModelExtensionUtils.getPresentableText(object);

        return text != null ? text : super.getPresentableText(object);
    }

    @Nullable
    @Override
    public PsiElement adjustElement(final PsiElement psiElement) {
        PsiElement element = NavBarModelExtensionUtils.adjustElement(psiElement);

        return element == null ? null : super.adjustElement(psiElement);
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/projectView/pycharm/PyCharmNavBarModelExtension.kt
package org.stepik.core.projectView.pycharm

import com.intellij.ide.navigationToolbar.DefaultNavBarExtension
import com.intellij.psi.PsiElement
import com.intellij.util.Processor
import org.stepik.core.projectView.NavBarModelExtensionUtils


class PyCharmNavBarModelExtension : DefaultNavBarExtension() {
    override fun getPresentableText(any: Any?): String? {
        val text = NavBarModelExtensionUtils.getPresentableText(any)
        return text ?: super.getPresentableText(any)
    }

    override fun adjustElement(psiElement: PsiElement): PsiElement? {
        NavBarModelExtensionUtils.adjustElement(psiElement) ?: return null
        return super.adjustElement(psiElement)
    }

    override fun processChildren(any: Any?, rootElement: Any?, processor: Processor<Any>?) = true
}
